src/mono/mono/mini/aot-compiler.c

```diff
@@ -6186,6 +6246,7 @@ is_direct_callable (MonoAotCompile *acfg, MonoMethod *method, MonoJumpInfo *patc
 	} else if ((patch_info->type == MONO_PATCH_INFO_ICALL_ADDR_CALL && patch_info->data.method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL)) {
 		if (acfg->aot_opts.direct_pinvoke)
 			return TRUE;
+		return mono_aot_direct_pinvoke_enabled_for_method (acfg, patch_info->data.method);
```

Comment:
I think you can bring in the old aot_opts.direct_pinvoke and in case where you check both that as well as calling the function, you can just call the function.

Reply:
Good catch! Thanks for the tip. Moved the `if (acfg->aot_opts.direct_pinvoke)` check into the beginning of `is_direct_pinvoke_specified_for_method`.